rotkehlchen/tasks/manager.py

```diff
@@ -269,61 +274,70 @@ def _maybe_schedule_xpub_derivation(self) -> Optional[list[gevent.Greenlet]]:
             ))
         return greenlets
 
-    def _maybe_query_ethereum_transactions(self) -> Optional[list[gevent.Greenlet]]:
-        """Schedules the ethereum transaction query task if enough time has passed"""
-        with self.database.conn.read_ctx() as cursor:
-            accounts = self.database.get_blockchain_accounts(cursor).eth
-            if len(accounts) == 0:
-                return None
-
-            now = ts_now()
-            dbevmtx = DBEvmTx(self.database)
-            queriable_accounts = []
-            for account in accounts:
-                _, end_ts = dbevmtx.get_queried_range(cursor, account, SupportedBlockchain.ETHEREUM)  # noqa: E501
-                if now - max(self.last_eth_tx_query_ts[account], end_ts) > ETH_TX_QUERY_FREQUENCY:
-                    queriable_accounts.append(account)
-
-        if len(queriable_accounts) == 0:
-            return None
+    def _maybe_query_evm_transactions(self) -> Optional[list[gevent.Greenlet]]:
+        """Schedules the evm transaction query task if enough time has passed"""
+        result = []
+        for blockchain in (SupportedBlockchain.ETHEREUM, SupportedBlockchain.OPTIMISM):
+            with self.database.conn.read_ctx() as cursor:
+                accounts = self.database.get_blockchain_accounts(cursor).get(blockchain)
+                if len(accounts) == 0:
+                    continue
+
+                now = ts_now()
+                dbevmtx = DBEvmTx(self.database)
+                queriable_accounts: list[ChecksumEvmAddress] = []
+                for account in accounts:
+                    _, end_ts = dbevmtx.get_queried_range(cursor, account, blockchain)  # type: ignore[arg-type]  # noqa: E501
+                    if now - max(self.last_evm_tx_query_ts[(account, blockchain.to_chain_id())], end_ts) > EVM_TX_QUERY_FREQUENCY:  # type: ignore[index]  # noqa: E501
+                        queriable_accounts.append(account)  # type: ignore[arg-type]
+
+            if len(queriable_accounts) == 0:
+                continue
 
-        ethereum = self.chains_aggregator.get_chain_manager(SupportedBlockchain.ETHEREUM)
-        address = random.choice(queriable_accounts)
-        task_name = f'Query ethereum transactions for {address}'
-        log.debug(f'Scheduling task to {task_name}')
-        self.last_eth_tx_query_ts[address] = now
-        return [self.greenlet_manager.spawn_and_track(
-            after_seconds=None,
-            task_name=task_name,
-            exception_is_error=True,
-            method=ethereum.transactions.single_address_query_transactions,
-            address=address,
-            start_ts=0,
-            end_ts=now,
-        )]
+            evm_manager = self.chains_aggregator.get_chain_manager(blockchain)  # type: ignore[arg-type]  # noqa: E501
+            address = random.choice(queriable_accounts)
+            task_name = f'Query {blockchain.to_chain_id().to_name()} transactions for {address}'
```

Comment:
It's the same thing

```suggestion
            task_name = f'Query {str(blockchain)} transactions for {address}'
```

We can even add a `name()` function to `SupportedBlockchain` if it would make it easier for you to understand